pkg/process/checks/process_nix_test.go

```diff
@@ -219,3 +148,282 @@ func TestBasicProcessMessages(t *testing.T) {
+			ctrProcs: []ctrProc{
+				{ctrID: "", pCounts: 3},
+				{ctrID: "1", pCounts: 40},
+				{ctrID: "2", pCounts: 110},
+				{ctrID: "3", pCounts: 80},
+				{ctrID: "4", pCounts: 10},
+				{ctrID: "5", pCounts: 40},
+				{ctrID: "6", pCounts: 20},
+				{ctrID: "7", pCounts: 90},
+			},
+			expectedBatches: []map[string]int{
+				{"": 3},
+				{"1": 40},
+				{"2": 110},
+				{"3": 80, "4": 10},
+				{"5": 40, "6": 20},
+				{"7": 90},
+			},
+			expectedCtrCount:  7,
+			expectedProcCount: 393,
+			maxSize:           10,
+			maxCtrProcSize:    100,
+			containerHostType: model.ContainerHostType_notSpecified,
+		},
+		{
+			testName: "container process over batch size",
+			ctrProcs: []ctrProc{
+				{ctrID: "1", pCounts: 24},
+				{ctrID: "2", pCounts: 45},
+				{ctrID: "3", pCounts: 209},
+				{ctrID: "4", pCounts: 30},
+				{ctrID: "5", pCounts: 1},
+				{ctrID: "6", pCounts: 1},
+				{ctrID: "7", pCounts: 30},
+			},
+			expectedBatches: []map[string]int{
+				{"1": 24, "2": 45, "4": 30, "5": 1},
+				{"3": 209},
+				{"6": 1, "7": 30},
+			},
+			expectedCtrCount:  7,
+			expectedProcCount: 340,
+			maxSize:           10,
+			maxCtrProcSize:    100,
+			containerHostType: model.ContainerHostType_notSpecified,
+		},
+	} {
+		t.Run(tc.testName, func(t *testing.T) {
+			networks := make(map[int32][]*model.Connection)
+			procs, ctrs := generateCtrProcs(tc.ctrProcs)
+			procsByPid := procsToHash(procs)
+
+			lastRun := time.Now().Add(-5 * time.Second)
+			syst1, syst2 := cpu.TimesStat{}, cpu.TimesStat{}
+			cfg := config.NewDefaultAgentConfig(false)
+			sysInfo := &model.SystemInfo{}
+			lastCtrRates := util.ExtractContainerRateMetric(ctrs)
+			cfg.MaxPerMessage = tc.maxSize
+			cfg.MaxCtrProcessesPerMessage = tc.maxCtrProcSize
+			cfg.ContainerHostType = tc.containerHostType
+
+			processes := fmtProcesses(cfg, procsByPid, procsByPid, ctrIDForPID(ctrs), syst2, syst1, lastRun, networks)
+			containers := fmtContainers(ctrs, lastCtrRates, lastRun)
+			messages, totalProcs, totalContainers := createProcCtrMessages(processes, containers, cfg, sysInfo, int32(i), "nid")
+
+			assert.Equal(t, tc.expectedProcCount, totalProcs)
+			assert.Equal(t, tc.expectedCtrCount, totalContainers)
+
+			// sort and verify messages
+			sortMsgs(messages)
+			verifyBatchedMsgs(t, cfg, tc.expectedBatches, messages)
+		})
+	}
+}
+
+// sortMsgs sorts the CollectorProc messages so they can be validated deterministically
+func sortMsgs(m []model.MessageBody) {
+	// sort the processes and containers of each message
+	for i := range m {
+		payload := m[i].(*model.CollectorProc)
+		sort.SliceStable(payload.Containers, func(i, j int) bool {
+			return payload.Containers[i].Id < payload.Containers[j].Id
+		})
+		sort.SliceStable(payload.Processes, func(i, j int) bool {
+			return payload.Processes[i].Pid < payload.Processes[j].Pid
+		})
+	}
+
+	// sort all the messages by containers then by PIDs
+	sort.SliceStable(m, func(i, j int) bool {
```

Comment:
Since the containers and processes inside a message are already sorted, can't we just compare the id of the first element between two messages?

Reply:
Thanks, I had the more complicated sort because I had some more complicated verification at previously.